src/micro_http/src/common/headers.rs

```diff
@@ -286,6 +312,174 @@ impl Headers {
+                                            // this mean that the client does not want identity
+                                            // and this is the only encoding we support so far.
+                                            Ok(nr) if nr == 0.0 => Err(RequestError::HeaderError(
+                                                HttpHeaderError::InvalidValue(
+                                                    "Accept-Encoding".to_string(),
+                                                    encoding.to_string(),
+                                                ),
+                                            )),
+                                            Ok(_) => Ok(Self::Identity),
+                                        },
+                                        _no_value => Err(RequestError::HeaderError(
+                                            HttpHeaderError::InvalidValue(
+                                                "Accept-Encoding".to_string(),
+                                                encoding.to_string(),
+                                            ),
+                                        )),
+                                    }?;
+                                    Ok(Self::Identity)
+                                }
+                                _identity_no_q_value => Ok(Self::Identity),
+                            }?;
+                            Ok(Self::Identity)
+                        }
+                        "*" => {
+                            let encoding_pair = headers_str.splitn(2, '=').collect::<Vec<&str>>();
+                            match encoding_pair.len() {
+                                2 => match encoding_pair[1].parse::<f32>() {
+                                    Err(_) => Err(RequestError::HeaderError(
+                                        HttpHeaderError::InvalidValue(
+                                            "Accept-Encoding".to_string(),
+                                            encoding.to_string(),
+                                        ),
+                                    )),
+
+                                    Ok(nr) if nr == 0.0 && !q_value_identity => Err(
+                                        RequestError::HeaderError(HttpHeaderError::InvalidValue(
+                                            "Accept-Encoding".to_string(),
+                                            encoding.to_string(),
+                                        )),
+                                    ),
+
+                                    Ok(_q_value) => Ok(Self::Identity),
+                                },
+                                _ => Err(RequestError::HeaderError(HttpHeaderError::InvalidValue(
+                                    "Accept-Encoding".to_string(),
+                                    encoding.to_string(),
+                                ))),
+                            }
```

Comment:
This seems very similar to what happens on the `2 =>` branch. Can we deduplicate this code ?